parcels/field.py

```diff
@@ -1101,13 +1108,17 @@ def data(self):
     @property
     def time(self):
         try:
-            time = getattr(self.dataset, self.dimensions['time'])
-            if isinstance(time[0], np.datetime64) or 'Unit' not in time.attrs:
-                return np.array(time)
-            time.attrs['units'] = time.attrs['Unit']
-            ds = xr.decode_cf(self.dataset)
-            time = getattr(ds, self.dimensions['time'])
-            time_arr = np.array(time)
-            return time_arr
+            time_da = getattr(self.dataset, self.dimensions['time'])
+            if self.dataset['decoded'] and 'Unit' not in time_da.attrs:
+                time = np.array(time_da)
+            else:
+                if 'units' not in time_da.attrs and 'Unit' in time_da.attrs:
+                    time_da.attrs['units'] = time_da.attrs['Unit']
+                ds = xr.Dataset({self.dimensions['time']: time_da})
+                ds = xr.decode_cf(ds)
+                time = np.array(getattr(ds, self.dimensions['time']))
+            if isinstance(time[0], datetime.datetime):
+                raise NotImplementedError('Parcels does currently only parses date ranging from 1678 AD to 2262 AD, which are stored by xarray as np.datetime64. A larger time range could be implemented on demand.')
```

Comment:
Slight grammatical improvement to error message: ''Parcels currently only parses dates ranging from 1678 AD to 2262 AD, which are stored by xarray as np.datetime64. If you need a wider date range, please open an Issue on the parcels github page."